tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/BindingPatternTest.java

```diff
@@ -0,0 +1,144 @@
+ * under the License.
+ */
+
+package io.ballerina.semantic.api.test;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.ArrayTypeSymbol;
+import io.ballerina.compiler.api.symbols.MapTypeSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.TypeDescKind;
+import io.ballerina.compiler.api.symbols.TypeSymbol;
+import io.ballerina.compiler.api.symbols.VariableSymbol;
+import io.ballerina.projects.Document;
+import io.ballerina.projects.Project;
+import io.ballerina.tools.text.LineRange;
+import org.ballerinalang.test.BCompileUtil;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.ballerina.compiler.api.symbols.TypeDescKind.ANY;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.ANYDATA;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.ARRAY;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.ERROR;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.FLOAT;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.INT;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.MAP;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.STRING;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.UNION;
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDefaultModulesSemanticModel;
+import static io.ballerina.semantic.api.test.util.SemanticAPITestUtils.getDocumentForSingleSource;
+import static io.ballerina.tools.text.LinePosition.from;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Test cases for binding patterns.
+ *
+ * @since 2.0.0
+ */
+public class BindingPatternTest {
+
+    private SemanticModel model;
+    private Document srcFile;
+
+    @BeforeClass
+    public void setup() {
+        Project project = BCompileUtil.loadProject("test-src/binding_patterns_test.bal");
+        model = getDefaultModulesSemanticModel(project);
+        srcFile = getDocumentForSingleSource(project);
+    }
+
+    @Test(dataProvider = "CBPPosProvider")
+    public void testCaptureBindingPattern(int line, int col, String name, TypeDescKind typeKind) {
+        Optional<Symbol> symbol = model.symbol(srcFile, from(line, col));
+        assertTrue(symbol.isPresent());
+        assertEquals(symbol.get().getName().get(), name);
+        assertEquals(((VariableSymbol) symbol.get()).typeDescriptor().typeKind(), typeKind);
+    }
+
+    @DataProvider(name = "CBPPosProvider")
+    public Object[][] getCBPPos() {
+        return new Object[][]{
+                {18, 8, "cbp1", INT},
+                {19, 8, "cbp2", STRING}
+        };
+    }
+
+    @Test
+    public void testWildcardBindingPattern() {
+        Optional<Symbol> symbol = model.symbol(srcFile, from(22, 4));
+//        assertTrue(symbol.isEmpty());
```

Comment:
Shall we remove this line if it is not required.

Reply:
Ah. Was expecting this to be null for the wild card binding pattern. But we seem to be creating a symbol for that. Will check with others whether we need to expose that or ignore it.